Source/WebKit/UIProcess/API/APIWebAuthenticationPanel.cpp

```diff
@@ -37,14 +37,26 @@ namespace API {
 using namespace WebCore;
 using namespace WebKit;
 
+class NullPanelClient final : public RefCounted<NullPanelClient>, public WebAuthenticationPanelClient {
```

Comment:
Why do we need this now? The new PR looks significantly worse than the earlier one. I think we should go back to the earlier iteration if someone was constructing a `WebAuthenticationPanelClient` (and not a subclass of `WebAuthenticationPanelClient`).

Reply:
I don't think this is a bad code. Interface should not be used as concrete class in my opinion so I agreed to change the code. But I don't have enough time to work with this code, I reverted this as you suggested.